src/index.js

```diff
@@ -168,10 +168,19 @@ class Service {
 
   // STILL NEED TO ADD params argument here.
   create(data) {
-    return this.table.insert(data).run()
-      .then(res => Object.assign({
-        id: (data.id ? data.id : res.generated_keys[0])
-      }, data));
+    const idField = this.id;
+    return this.table.insert(data).run().then(function (res) {
+      if (data[idField]) {
+        if (res.errors) {
+          return Promise.reject(new errors.Conflict('Duplicate primary key'));
```

Comment:
BTW, this `duplicate primary key` exception was there even before I came around, but it just wasn't handled:

A user might have passed their own `id` in the data, and if there was a collision, data just wouldn't be inserted.